topology/probes/k8s/client.go

```diff
@@ -80,6 +81,41 @@ type kubeCache struct {
 	stopController chan (struct{})
 }
 
+func (c *kubeCache) list() []interface{} {
+	return c.cache.List()
+}
+
+func getStringField(obj interface{}, field string) string {
+	r := reflect.ValueOf(obj)
+	f := reflect.Indirect(r).FieldByName(field)
+	return f.String()
+}
+
+func (c *kubeCache) listByNamespace(namespace string) (objList []interface{}) {
```

Comment:
It seems this function is used only once, and the type is perfectly known so there is no reason to use a slow reflection to get a simple field.

Reply:
Is is going to be used by other throughout the code base once we create graph.Node to represent namespaces as in:

```
$ kubectl get namespaces
NAME          STATUS    AGE
default       Active    3d
kube-public   Active    3d
kube-system   Active    3d
```

Reply:
I suggest we do this modification when we'll really need it.

Reply:
ok, will be addressed in update to: `k8s: networkPolicy chain namespace-filter then pod-selection-filter`